src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +88,9 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                //i use MvcOption.requireHttpsPermanent value if Permanent parameter is null
+                _permanent = _permanent ?? optionsAccessor.Value.RequireHttpsPermanent;
```

Comment:
(And yes, there is such a thing as reloadable configuration and options, but MVC doesn't use it, so that isn't an actual concern here.)